herddb-core/src/main/java/herddb/sql/DDLSQLPlanner.java

```diff
@@ -365,7 +364,14 @@ private Statement buildCreateTableStatement(String defaultTableSpace, CreateTabl
                     if (auto_increment && primaryKey.contains(cf.getColumnName())) {
                         tablebuilder.primaryKey(columnName, auto_increment);
                     }
+
+                    if(String.join("_",columnSpecs).equals("NOT_NULL")) {
+                        type = ColumnTypes.getNonNullTypeForPrimitiveType(type);
```

Comment:
we should throw an error in case of a type which does not support NOT NULL
and we have to add an unit test around this case.

we can't let users try to create a constraint but this specs won't reflect the actual type of the column

Reply:
Good idea. Will add UT for this.

Reply:
@eolivelli Addressed changes and added UT